### PR TITLE
Update Go imports for etcdadm repos and patched CAPI fork

### PR DESCRIPTION
**Pull request by abhay-krishna**
This PR updates the Go imports to use the etcdadm controller and bootstrap provider repos under AWS org, and the CAPI fork of `abhay-krishna` instead of `mrajashree`, that is no longer actively maintained.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/api/v1alpha1/thirdparty/tinkerbell/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -19,8 +19,6 @@
 
 package tinkerbell
 
-import ()
```
Comment (g-gaston) [inline pkg/api/v1alpha1/thirdparty/tinkerbell/zz_generated.deepcopy.go:22]: was this due to a kubebuilder change?

File: pkg/api/v1alpha1/zz_generated.deepcopy.go [hunk 2]
```diff
@@ -20,10 +20,11 @@
 package v1alpha1
 
 import (
-	apiv1beta1 "github.com/aws/eks-anywhere/pkg/providers/snow/api/v1beta1"
 	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/cluster-api/api/v1beta1"
+
+	apiv1beta1 "github.com/aws/eks-anywhere/pkg/providers/snow/api/v1beta1"
```
Comment (g-gaston) [inline pkg/api/v1alpha1/zz_generated.deepcopy.go:27]: same question with this, kubebuilder?

File: release/go.sum [hunk 3]
```diff
@@ -444,6 +444,7 @@ github.com/hashicorp/errwrap v1.0.0/go.mod h1:YH+1FKiLXxHSkmPseP+kNlulaMuP3n2brv
 github.com/hashicorp/go-cleanhttp v0.5.1/go.mod h1:JpRdi6/HCYpAwUzNwuwqhbovhLtngrth3wmdIIUrZ80=
 github.com/hashicorp/go-immutable-radix v1.0.0/go.mod h1:0y9vanUI8NX6FsYoO3zeMjhV/C5i9g4Q3DwcSNZ4P60=
 github.com/hashicorp/go-msgpack v0.5.3/go.mod h1:ahLV/dePpqEmjfWmKiqvPkv/twdG7iPBM1vqhUKIvfM=
+github.com/hashicorp/go-multierror v1.0.0/go.mod h1:dHtQlpGsu+cZNNAkkCN/P3hoUDHhCYQXV3UM06sGGrk=
```
Comment (g-gaston) [inline release/go.sum:447]: curious why the release go.sum changed without changing the go.mod?

File: pkg/api/v1alpha1/awsiamconfig_test.go [hunk 4]
```diff
@@ -10,7 +10,7 @@ import (
 func TestGetAndValidateAWSIamConfig(t *testing.T) {
 	c := &Cluster{
 		ObjectMeta: metav1.ObjectMeta{
-			ClusterName: "eksa-unit-test-cluster",
+			Name: "eksa-unit-test-cluster",
```
Comment (g-gaston) [inline pkg/api/v1alpha1/awsiamconfig_test.go:13]: curious what happened here? This seems to be our own api struct?

File: go.mod [hunk 5]
```diff
@@ -19,37 +20,36 @@ require (
...
+	golang.org/x/sys v0.0.0-20220520151302-bc2c85ada10a
+	gopkg.in/ini.v1 v1.66.4
 	gopkg.in/square/go-jose.v2 v2.6.0
 	gopkg.in/yaml.v2 v2.4.0
 	gopkg.in/yaml.v3 v3.0.0
-	k8s.io/api v0.23.4
-	k8s.io/apimachinery v0.23.4
-	k8s.io/client-go v0.23.4
+	k8s.io/api v0.24.2
+	k8s.io/apimachinery v0.24.2
```
Comment (g-gaston) [inline go.mod:40]: curious, these changes come from updating capi? or we are just updating everything at once

**Review by g-gaston — COMMENTED** [review]
left a few comments about things that picked my attention but no major concerns

File: pkg/api/v1alpha1/zz_generated.deepcopy.go (hunk 2 above)
Comment (abhay-krishna) [reply pkg/api/v1alpha1/zz_generated.deepcopy.go:27]: this was due to `gci`, that fixes imports

File: pkg/api/v1alpha1/thirdparty/tinkerbell/zz_generated.deepcopy.go (hunk 1 above)
Comment (abhay-krishna) [reply pkg/api/v1alpha1/thirdparty/tinkerbell/zz_generated.deepcopy.go:22]: this was due to `gci` tool, that fixes imports.

Comment (g-gaston) [reply pkg/api/v1alpha1/thirdparty/tinkerbell/zz_generated.deepcopy.go:22]: Ah yeah. Could we revert it? We shouldn't change autogenerated files. The linter is already configured to omit these files.

**Comment by TerryHowe** [comment]
Looks very stale

**Comment by g-gaston** [comment]
Closing, I believe this has already been updated separately

**Pull request closed without merging**